src/plugins/imagery/mixins/imageryData.js

```diff
@@ -145,6 +147,25 @@ export default {
                 this.imageHistory = imagery;
             }
         },
+        clearData(domainObjectToClear) {
```

Comment:
should we add a test for this method?

Reply:
I might need some guidance on the best way to access the method for testing purposes. Its not available on the ImageView or ImageViewProvider because it's used as a mixin in ImageryView.vue.

Reply:
It might be easiest to add a test to [the spec for the imagery view itself](https://github.com/nasa/openmct/blob/e14b7cd0e28f0ca6f1ffbdaeca0521d33c9b024c/src/plugins/imagery/pluginSpec.js#L354).

You could use a query selector to count the number of images first, then invoke the clearData action, then verify that the number of images is now 0.

eg.
``` JavaScript
expect(parent.querySelectorAll('c-imagery__thumb').length).not.toBe(0);
const clearDataAction = openmct.actions.getAction('clearData');
clearDataAction.invoke();
await Vue.nextTick()
expect(parent.querySelectorAll('c-imagery__thumb').length).toBe(0));
```

Above code is untested and should be considered pseudo-code :)

Reply:
```js 
it('clearData is called when Clear Data for Object is selected', (done) => {
     expect(parent.querySelectorAll('.c-imagery__thumb').length).not.toBe(0);
     openmct.objectViews.on('clearData', async (domainObject) => {
           await Vue.nextTick();
            expect(parent.querySelectorAll('.c-imagery__thumb').length).toBe(0);
            done();
        });
     clearDataAction.invoke([imageryObject]);
});
```

I've added the above test but ran into a few unexpected results. 

* In local, the `                expect(parent.querySelectorAll('.c-imagery__thumb').length).toBe(0);
` clause will fail due to the requestHistory being invoked after the clearData method is triggered; so the imageHistory is emptied but then immediately refilled at the end of clearData. 
* When testing this against CI, the ` expect(parent.querySelectorAll('.c-imagery__thumb').length).not.toBe(0);` clause fails, which  was unexpected and might be a result of a difference in example.imagery in the CI environment.

Reply:
@michaelrogers I know we talked already, but just putting this here for posterity:
I think the issue is the fake telemetry generator in the test always generates new images, and never lets the imageHistory clear out. In practice, the `clearDataAction` [explicitly relies on the imageHistory clearing out](https://github.com/nasa/openmct/pull/4733/files#diff-92f36d52f254f953f1bd763560ada01ed5dd94349e97a0eef97f22ef165df96aR172), so I think this should be fine. In this test, we should restub the telemetry generator so the imageHistory actually empties properly.